tests/unit/conftest.py

```diff
@@ -17,3 +35,17 @@ def harness_fixture():
     yield harness
 
     harness.cleanup()
+
+
+@pytest.fixture(scope="function", name="harness_with_agent_relation")
+def harness_with_agent_relation_fixture(harness: Harness, agent_relation_data: dict) -> Harness:
+    """Harness with agent relation to jenkins-k8s.
+
+    Args:
+        harness the default testing harness.
+
+    Returns:
+        The harness with agent relation established.
+    """
+    harness.add_relation(AGENT_RELATION, "jenkins-k8s", unit_data=agent_relation_data)
```

Comment:
Thanks for the feedback, Can you let me know in more details how this can cause complexity in synapse? This is also a fairly simple charm compared to synapse and we probably won't have to make much modification to the charm code so I wonder if we'll face the same complexity problem as synapse.

Reply:
Seeing as this is a fairly complex topic, let's continue the discussion outside of this PR and I can create a follow-up PR if needed. I'll mark this conversation as resolved